From 5fee507fef1a322efabceee6f938195795d90eea Mon Sep 17 00:00:00 2001
From: Pete Walter <pwalter@fedoraproject.org>
Date: Wed, 15 Jan 2025 23:34:08 +0000
Subject: [PATCH] ext: Fix the build with system libgit2 1.9.0

Update version.h parsing regexes for 1.9.0 and fix compilation test by
including git2.h instead of directly including git2/version.h, which is
no longer supported.
---
 ext/rugged/extconf.rb | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/ext/rugged/extconf.rb b/ext/rugged/extconf.rb
index 7dbd2b3b9..5b1a21c19 100644
--- a/ext/rugged/extconf.rb
+++ b/ext/rugged/extconf.rb
@@ -71,12 +71,12 @@ def self.run_cmake(timeout, args)
   major = minor = nil
 
   File.readlines(File.join(LIBGIT2_DIR, "include", "git2", "version.h")).each do |line|
-    if !major && (matches = line.match(/^#define LIBGIT2_VER_MAJOR\s+([0-9]+)$/))
+    if !major && (matches = line.match(/^#define LIBGIT2_VERSION_MAJOR\s+([0-9]+)$/))
       major = matches[1]
       next
     end
 
-    if !minor && (matches = line.match(/^#define LIBGIT2_VER_MINOR\s+([0-9]+)$/))
+    if !minor && (matches = line.match(/^#define LIBGIT2_VERSION_MINOR\s+([0-9]+)$/))
       minor = matches[1]
       next
     end
@@ -85,9 +85,9 @@ def self.run_cmake(timeout, args)
   end
 
   try_compile(<<-SRC) or abort "libgit2 version is not compatible, expected ~> #{major}.#{minor}.0"
-#include <git2/version.h>
+#include <git2.h>
 
-#if LIBGIT2_VER_MAJOR != #{major} || LIBGIT2_VER_MINOR != #{minor}
+#if LIBGIT2_VERSION_MAJOR != #{major} || LIBGIT2_VERSION_MINOR != #{minor}
 #error libgit2 version is not compatible
 #endif
   SRC
